crates/biome_fs/src/fs/os.rs

```diff
@@ -83,9 +83,17 @@ impl FileSystem for OsFileSystem {
         path.is_file()
     }
 
-    fn resolve_configuration(&self, specifier: &str) -> Result<Resolution, ResolveError> {
-        self.configuration_resolver
-            .resolve(self.working_directory().unwrap(), specifier)
```

Comment:
This is the root cause of the issue. The default working directory in LSP mode is the directory where the editor executable is at.